sqlglot/dialects/clickhouse.py

```diff
@@ -476,20 +476,20 @@ def _parse_function(
 
                 if params:
                     if parts and parts[1]:
-                        return self.expression(
+                        func = self.expression(
                             exp.CombinedParameterizedAgg,
                             this=func.this,
                             expressions=func.expressions,
                             params=params,
                             parts=parts,
                         )
-                    return self.expression(
+                    func = self.expression(
                         exp.ParameterizedAgg,
                         this=func.this,
                         expressions=func.expressions,
                         params=params,
                     )
-
+                    return self._parse_window(func)
                 if parts:
                     if parts[1]:
                         return self.expression(
```

Comment:
What about this and L501? Shouldn't we call `_parse_window` once at the end of this method?

Reply:
I was thinking about this but I think this will be a no-op for the case of `parts`, which only serves the purpose of  splitting the aggfunc name from it's sufix. If we don't have `params`, then there is nothing standing between the function call and the window spec. Thus, this can be fully parsed by `parser.py::_parse_function()`. 

Also, this whole branch is only accessed if the `parser.py__parse_function()` call has returned an anonymous function, so if a window was parsed too we will never reach this point. Sidenote: If i'm not missing something, that means that another side-effect was found; CH can't tag `exp.Anonymous` functions as `exp.CombinedParameterizedAgg / exp.ParameterizedAgg` if they're parsed and wrapped in a `exp.Window`, maybe we can check `window.this` and substitute it after the fact.

Reply:
> I was thinking about this but I think this will be a no-op

Sounds good, I think I see what you mean.

> CH can't tag exp.Anonymous functions as `exp.CombinedAggFunc` / `exp.AnonymousAggFunc` if they're parsed and wrapped in a `exp.Window`

Good catch, can we fix this?

Reply:
There started being too much noise so I tried to refactor things and hopefully make some conditions clearer to see.

Is this version preferred or should I merge the sidenote fix with the old code?